snarkos/network/mod.rs

```diff
@@ -234,23 +234,28 @@ pub async fn handle_listener<N: Network>(listener: TcpListener, ledger: Arc<Ledg
 
 /// Send a ping to all peers every 10 seconds.
 pub async fn send_pings<N: Network>(ledger: Arc<Ledger<N>>) -> Result<(), Box<dyn std::error::Error>> {
-    loop {
-        thread::sleep(time::Duration::from_secs(10));
+    tokio::spawn(async move {
```

Comment:
we should return this task's handle from this method; even if we don't track task handles yet, we will want to be doing it shortly

Reply:
as a reference, we're already doing this with the HTTP server's tasks: https://github.com/AleoHQ/snarkOS/blob/1f6c46327ea9db61f5b8ea577645a976c2c8b9c1/snarkos/ledger/server.rs#L72

Reply:
done in https://github.com/AleoHQ/snarkOS/pull/1884, which is a follow-up to this PR